artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQMessageProducer.java

```diff
@@ -533,6 +496,52 @@ private void doSendx(ActiveMQDestination destination,
       }
    }
 
+   private void checkDestination(ActiveMQDestination destination,
+                                 SimpleString address,
+                                 ClientSession clientSession) throws JMSException {
+      if (!connection.containsKnownDestination(address)) {
+         try {
+            ClientSession.AddressQuery query = clientSession.addressQuery(address);
+
+            if (!query.isExists()) {
+               checkQueue(destination, address, clientSession, query);
```

Comment:
this is missing added the destination (another bug introduced by the original change)

Reply:
e.g. if we are succesfull, then the address becomes known right?

Reply:
it would always add the end, but I'm rewriting the logic now